src/utility/PricingStrategy.js

```diff
@@ -0,0 +1,96 @@
+export class PricingStrategy {
+  constructor(pricingJSON) {
+    let pricingData = JSON.parse(pricingJSON);
+    this.priceModel = pricingData.price_model;
+    if (this.priceModel === "fixed_price") {
+      this.pricingModel = new FixedPricing(pricingData);
+    } else if (this.priceModel === "fixed_price_per_method") {
+      this.pricingModel = new MethodPricing(pricingData);
+    }
+  }
+
+  getPriceModel() {
+    return this.priceModel;
+  }
+
+  getPriceInCogs(serviceName, methodName) {
+    return this.pricingModel.getPriceInCogs(serviceName, methodName);
+  }
+
+  getPriceInAGI(serviceName, methodName) {
+    return this.pricingModel.getPriceInAGI(serviceName, methodName);
+  }
+
+  getMaxPriceInCogs() {
+    return this.pricingModel.getMaxPriceInCogs();
+  }
+
+  getMaxPriceInAGI() {
+    return AGIUtils.inAGI(this.pricingModel.getMaxPriceInCogs());
+  }
+}
+
+class AGIUtils {
+  static toDecimal(agi) {
+    return (agi / 100000000).toFixed(8);
+  }
+
+  static inAGI(cogs) {
```

Comment:
Do we need both `toDecimal` and `inAGI` method. Both seem to be doing the same thing.

Reply:
Done